authentik/providers/oauth2/views/authorize.py

```diff
@@ -238,6 +256,131 @@ def create_code(self, request: HttpRequest) -> AuthorizationCode:
 
         return code
 
+class OAuthPostFulfillmentStage(ChallengeStageView):
```

Comment:
So this extra stage view shouldn't be required, imo it makes more sense to hook in here https://github.com/goauthentik/authentik/blob/master/authentik/providers/oauth2/views/authorize.py#L248, rename `redirect` to `get_response` and have it check the parameters to either return an Autosubmit Challenge or a redirect

Reply:
Yep, I agree, this was just a way of testing the concept on my part for now. I'll have a look at integrating it a bit better later today.

Reply:
Right, had some more free time this afternoon. I've removed the separate class and moved the FORM_POST handling to create_response_uri() and redirect() respectively

I'll have to spend some more time going through a couple of more scenarios where form_post might want a non-implicit response, I've only been testing with an application of ours that has a very specific capability for OIDC (Azure) so far.

https://github.com/scheibling/authentik/blob/b7fd2b38935ed63a2bd1214f6c41d6ac2e6485b9/authentik/providers/oauth2/views/authorize.py#L265-L282

Reply:
Looked at it with a little more sleep behind me, I'm going to work on the create_response_uri-part a bit more since the current code doesn't really take any of the response_type parameters into account. 

Form_post in particular can be requested with just the code and id_token parts by themselves, I'm gonna read up a bit on that part of the protocol when I get into work tomorrow. 

The original version was written as if fragment always just creates an implicit response with just an id_token and  query only provides a code, couldn't there be other combinations on those as well? 

I'll be back with an update once I'm up to speed with the latest!

Reply:
Nope, apparently the code that's already in create_implicit_response works fine for form_post as well!